src/player.cpp

```diff
@@ -14442,40 +14442,76 @@ std::vector<mission*> player::get_failed_missions() const
     return failed_missions;
 }
 
-void player::print_encumbrance(WINDOW *win, int min, int max, int line) const
+encumbrance_data player::get_encumbrance( size_t i ) const
+{
+    encumbrance_data e;
+    e.iBodyTempInt = ( temp_conv[i] / 100.0 ) * 2 - 100; // Scale of -100 to +100
+    e.iEnc = encumb( bp_aBodyPart[i], e.iLayers, e.iArmorEnc );
+    return e;
+}
+
+void player::print_encumbrance( WINDOW *win, int line ) const
 {
-    // initialize these once, and only once
-    static std::string asText[] = {_("Torso"), _("Head"), _("Eyes"), _("Mouth"), _("L. Arm"), _("R. Arm"),
-                             _("L. Hand"), _("R. Hand"), _("L. Leg"), _("R. Leg"), _("L. Foot"),
-                             _("R. Foot")};
-    static body_part aBodyPart[] = {bp_torso, bp_head, bp_eyes, bp_mouth, bp_arm_l, bp_arm_r, bp_hand_l,
-                             bp_hand_r, bp_leg_l, bp_leg_r, bp_foot_l, bp_foot_r};
-    int iEnc, iArmorEnc, iBodyTempInt;
-    double iLayers;
+    int height, width;
+    getmaxyx( win, height, width );
+
+    // fill a set with the indices of the body parts to display
+    int l = std::max( 0, line );
+    std::set<int> parts;
+    // check and optionally enqueue l+0, l-1, l+1, l-2, l+2, ...
+    int o = 0; // offset from l
+    int skip[2] = {}; // how far to skip on next neg/pos jump
+    do {
+        if ( !skip[o > 0] && l + o >= 0 && l + o < num_bp ) { // l+o is in bounds
```

Comment:
I was going to do some light restyling and merge it before I got here, but this is unreadable, l and o need to have meaningful names.